### ISSUE TITLE
MacOS release needs notarization

### ISSUE THREAD

**Issue by zemien** [rnwood/smtp4dev#323]
Hi, I saw your request to help test the MacOS release so here's the outcome:

TL/DR: `dotnet tools` approach works really well and should be the recommended approach for MacOS if they can't use the Linux docker container approach.

The .zip releases can't run properly on newer MacOS's because it hasn't been [notarized by Apple](https://developer.apple.com/documentation/xcode/notarizing_macos_software_before_distribution). I think you need an Apple developer account for that. I get a warning that it won't run binaries:
![image](https://user-images.githubusercontent.com/8407032/78769990-23e4fa00-79e2-11ea-89f5-3d43c79d24e6.png)

I can go into Security & Privacy to "Allow Anyway":
![image](https://user-images.githubusercontent.com/8407032/78770087-424af580-79e2-11ea-80ea-9f78bc402601.png)

But I continue to be prompted with warnings for **each and every .dll**! And I can only Allow Anyway one .dll at a time, so it becomes an endless whack-a-button game:
![image](https://user-images.githubusercontent.com/8407032/78770233-673f6880-79e2-11ea-9f25-59a6fa2c3d09.png)

I was interested to see what would happen if I allowed all of it, but I gave up after about the 20th allow.

The dotnet tools approach on the other hand worked well, and I didn't get any security warnings. It installed as an executable in the tools folder:
![image](https://user-images.githubusercontent.com/8407032/78770520-c7cea580-79e2-11ea-8dc7-23c2a173bc73.png)

![image](https://user-images.githubusercontent.com/8407032/78770430-ab326d80-79e2-11ea-991d-a3d1dc1f138d.png)

Hopefully this gives you enough information to either:

* notarize the MacOS release
* push MacOS users to use dotnet tools or the Linux container

**Comment by rnwood** [comment]
@zemien Thank you for the feedback on this.

**Comment by rnwood** [comment]
Help wanted please from someone with access to a MacOS machine.

**Comment by zemien** [comment]
I can help, what do you need?

**Comment by rnwood** [comment]
Help still wanted if anyone can offer it. 

Currently no macos release is generated because of previous feedback that it's better to use the framework dependent version (since that can apparently run without signing of the smtp4dev bits).

Can someone in the know and with a Mac please confirm if this is still the latest situation. Would a return of the macos version that doesn't require you to install .net runtime first be useful and does the lack of it prevent you from using smtp4dev?

**Comment by zemien** [comment]
Hi I did some reading and the rules have only tightened around signing: https://learn.microsoft.com/en-us/dotnet/core/install/macos-notarization-issues

It depends on your target MacOS audience. I came across this as a .Net developer needing a SMTP test server. So I already had `dotnet tools` installed and used it. I don’t think developers of any other framework (Node, Go, Python) would balk at having to install the dotnet sdk just to run this. It’s still easier than any alternative to have a functioning SMTP test server.

Unless you are an active Mac OS app publisher, I don’t think it’s worth the annual $99 USD to join the Apple Developer Program so that you can notarize the self-contained app that doesn’t require .Net runtime.

My recommendation? Update wiki to include steps for Mac users to install .net sdk and then the dotnet global tools command

**Comment by rnwood** [comment]
Thanks. I appreciate your recommendations on this. This is the direction I had in mind.

As well as updating the wiki, the table of downloads that goes into each GitHub release automatically will also be updated. We should add a link here to the other distribution methods in general to boost awareness of the dotnet tool and docker options.

**Comment by rnwood** [comment]
Implemented in #1734
